src/libpmem2/map_ravl.h

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * map_ravl.h -- internal definitions for map_ravl
+ */
+
+#ifndef MAP_RAVL_H
+#define MAP_RAVL_H
+
+#include "libpmem2.h"
+
+struct map_ravl;
```

Comment:
This is an interval tree. I'll leave it up to you, but I think it would be better to create a generalized ravl_interval api (that takes any two points on the number line) instead of a specialized one for maps.

I'm also not a fan of putting locks on the inside of data structures.

Reply:
`/*
 * ravl_interval - structure holding two points on a number line
 *                 representing a mapped space
 */
struct ravl_interval {
	size_t l_point;
	size_t r_point;
}`
Did you mean something like that?
Where l_point represents base address and r_point represents base address + mapping size.
And then instead of operating on the pmem2_map we should work on those points.

Reply:
```c++
/*
 * ravl_interval - structure holding two points on a number line
 *                 representing a mapped space
 */
struct ravl_interval {
	size_t l_point;
	size_t r_point;
}
```
Did you mean something like that?
Where l_point represents base address and r_point represents base address + mapping size.
And then instead of operating on the pmem2_map we should work on those points.

Reply:
Yes. Consider also the possibility of overlapping intervals (which was not accounted for in the existing implementation) - if you decide against implementing them, an attempt at an insert of an overlapping range should fail.

As a side not, I suggest to use `min` and `max` instead of left and right.